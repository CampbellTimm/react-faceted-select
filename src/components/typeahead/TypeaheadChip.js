import React from 'react';
import PropTypes from 'prop-types';

class TypeaheadChip extends React.Component {
    static propTypes = {
        children: PropTypes.object,
        onRemove: PropTypes.func,
    };

    handleClick = (event) =>{
        this.props.onRemove(this.props.children);
        event.preventDefault();
    };

    makeCloseButton = () => {
        if (!this.props.onRemove) {
            return '';
        }
        return (
            <a className="typeahead-chip-close" href="#" onClick={this.handleClick}>&#x00d7;</a>
        );
    };

    render() {
        const {category, operator, value} = this.props.children;
        return (
            <div className="typeahead-chip">
                <span className="chip-category">{category}</span>
                <span className="chip-operator">{operator}</span>
                <span className="chip-value">{value}</span>
                {this.makeCloseButton()}
            </div>
        );
    }
}

export default TypeaheadChip;