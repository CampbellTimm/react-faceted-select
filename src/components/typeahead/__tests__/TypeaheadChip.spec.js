import React from 'react';
import { shallow } from 'enzyme';
import TypeaheadChip from '../TypeaheadChip';

describe('TypeaheadChip', () => {

    it('should call onRemove when delete button clicked', () => {
        const onRemove = jest.fn();
        const preventDefault = jest.fn();

        const component = shallow(
            <TypeaheadChip
                children={{
                    category: "Name",
                    operator: "contains",
                    value: "John"
                }}
                onRemove={onRemove}
            />
        );

        component
            .find('a.typeahead-chip-close')
            .simulate('click', { preventDefault });

        expect(onRemove).toHaveBeenCalled();
        expect(component).toMatchSnapshot();
    });
});