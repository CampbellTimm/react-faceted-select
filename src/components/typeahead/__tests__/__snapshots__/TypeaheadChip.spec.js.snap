// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeaheadChip should call onRemove when delete button clicked 1`] = `
<div
  className="typeahead-chip"
>
  <span
    className="chip-category"
  >
    Name
  </span>
  <span
    className="chip-operator"
  >
    contains
  </span>
  <span
    className="chip-value"
  >
    John
  </span>
  <a
    className="typeahead-chip-close"
    href="#"
    onClick={[Function]}
  >
    Ã—
  </a>
</div>
`;
